lib/rubocop/cop/registry.rb

```diff
@@ -23,34 +23,28 @@ def initialize(name, origin, badges)
     # Registry that tracks all cops by their badge and department.
     class Registry
       def initialize(cops = [], options = {})
-        @registry = {}
-        @departments = {}
-        @cops_by_cop_name = Hash.new { |hash, key| hash[key] = [] }
-
+        @enlisted = []
         cops.each { |cop| enlist(cop) }
         @options = options
       end
 
       def enlist(cop)
-        @registry[cop.badge] = cop
-        @departments[cop.department] ||= []
-        @departments[cop.department] << cop
-        @cops_by_cop_name[cop.cop_name] << cop
+        @enlisted << cop
       end
 
       # @return [Array<Symbol>] list of departments for current cops.
       def departments
-        @departments.keys
+        staff.keys
       end
 
       # @return [Registry] Cops for that specific department.
       def with_department(department)
-        with(@departments.fetch(department, []))
+        with(staff.fetch(department, []))
```

Comment:
Since `staff` is a `Hash.new { |hash, key| hash[key] = [] }` it might make sense to get rid of `fetch`.